pkg/querier/frontend/frontend.go

```diff
@@ -179,8 +179,19 @@ func (f *Frontend) Handler() http.Handler {
 }
 
 func (f *Frontend) handle(w http.ResponseWriter, r *http.Request) {
-
 	startTime := time.Now()
+
+	// to parse form we need to be sure
+	// that roundtriper request gets not read reader
+	var buf bytes.Buffer
+	r.Body = ioutil.NopCloser(io.TeeReader(r.Body, &buf))
+	// Ensure the form has been parsed so all the parameters are present
+	err := r.ParseForm()
+	if err != nil {
+		level.Warn(util.WithContext(r.Context(), f.log)).Log("msg", "unable to parse form for request", "err", err)
+	}
+	r.Body = ioutil.NopCloser(&buf)
+
```

Comment:
What about moving `startTime := time.Now()` to this line?

Reply:
Good point 👍